source/world_builder/utilities.cc

```diff
@@ -427,9 +434,166 @@ namespace WorldBuilder
+                    {
+                      check_point_surface_2d_temp_ob[0]-= 2*Consts::PI;
+                    }
+                }
+
+              const Point<2> AB_normal_ob = closest_point_on_curve.normal*closest_point_on_line_2d.distance(reference_point);//*AB.norm();
+              const Point<2> local_reference_point_ob = AB_normal_ob*1.+closest_point_on_line_2d;
+              const bool reference_normal_on_side_of_line_ob =  (closest_point_on_line_2d-local_reference_point_ob).norm_square() < (check_point_surface_2d_temp_ob-local_reference_point_ob).norm_square();
+              const bool reference_point_on_side_of_line_ob =  (point_list[point_list.size()-1][0] - point_list[0][0])*(reference_point[1] - point_list[0][1]) - (reference_point[0] - point_list[0][0])*(point_list[point_list.size()-1][1] - point_list[0][1]) < 0.;
+
+              double line_factor;
+              double old_dist;
+              double new_dist;
+              Point<2> iterable_check_point_surface_2d = check_point_surface_2d;
+              Objects::ClosestPointOnCurve iterable_closest_point_on_curve = bezier_curve.closest_point_on_curve_segment(iterable_check_point_surface_2d);
+              bool converged = false;
+
+              if (reference_normal_on_side_of_line_ob != reference_point_on_side_of_line_ob)
+                {
+                  // We want to take the current checkpoint and move it along the obliquity vector using a parameterized line
+                  // towards the bezier curve until we find a point on the bezier curve that intersects this parameterized line.
+                  for (unsigned int i = 0; i < 100; ++i)
+                    {
+                      // We will use the distance of the check point to the current "closest point on curve" to scale the distance
+                      // traveled along the parameterized line.
+                      old_dist = iterable_closest_point_on_curve.distance / 1.1;
+
+                      // If the sign of the dot product of the obliquity vector and the normal vector of the closest point on
+                      // the curve is positive, then we want to move along the obliquity vector in the negative direction,
+                      // otherwise in the positive direction.
+                      line_factor = obliquity_vector * iterable_closest_point_on_curve.normal > 0 ? -1.0 : 1.0;
+
+                      // Parameterize a line through the checkpoint along the obliquity vector.
+                      Point<2> parameterized_line(iterable_check_point_surface_2d[0] + line_factor * old_dist * obliquity_vector[0],
+                                                  iterable_check_point_surface_2d[1] + line_factor * old_dist * obliquity_vector[1],
+                                                  natural_coordinate_system);
+
+                      // Check where the closest point on the bezier curve is to the new check point (after moving along the
```

Comment:
```suggestion
                      // Check where the closest point on the Bezier curve is relative to the new check point (after moving along the
```
relative to? with respect to?